### DISCUSSION TITLE
[FEATURE] Make log level in kvssink configurable.

### DISCUSSION THREAD

**Discussion by mateuszboryn** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#1182] [Q&A]
**Is your feature request related to a problem? Please describe.**
Currently, kvssink GST component has hardcoded loglevel to DEBUG. 
https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/blame/master/src/gstreamer/gstkvssink.cpp#L761

It is annoying to see multiple DEBUG messages in long-running GST pipeline.

**Describe the solution you'd like**
Kindly please make log4cpp logging appender push to gstreamer logging.
This way, kvssink would have gstreamer compatible logging.

**Describe alternatives you've considered**
1. Hardcode higher level of logging instead of DEBUG (INFO, WARN?) - quick fix, but hiding any potential issues permanently.
2. Set log level there, with `if(GST_LOG_LEVEL==GST_DEBUG) { LOG_CONFIGURE_STDOUT("DEBUG");} else if(GST_LOG_LEVEL==GST_INFO) {LOG_CONFIGURE_STDOUT("INFO");} ` statement during initialization - better one, but logging output is still fully controlled by log4cpp appender.
3. Rely on log4cpp config (for GST pipelines it would be yet another place to configure logging, which is inconvenient for gstreamer users)

**Additional context**
none

**Comment by disa6302** [comment]
@mateuszboryn ,

The log level should be configurable. The SDK uses the kvs_log_configuration file to set the log level. (https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/blob/v3.4.1/src/gstreamer/gstkvssink.cpp#L1583-L1590). This is where you set the log level if using the SDK provided log file: https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/blob/v3.4.1/kvs_log_configuration#L1

You could also provide your own log config file by using the log-config property. 


Am I missing something here or does this not work for you?

**Reply by mateuszboryn** [reply]
Well, I'm aware of that, however, please take a closer look at the initialization of kvssink I pointed out. There is `LOG_CONFIGURE_STDOUT("DEBUG");` hardcoded which overrides anything I set in kvs_log_configuration file. That's the point.

**Reply by colonelwatch** [reply]
@mateuszboryn Hello, I found your thread after trying to solve the same issue. I'm not sure if the above response has resolved it for you already, but it was in fact resolved by setting the log level in `kvs_log_configuration`. However, we discovered that the SDK-provided log file isn't necessarily used by default. By default, it uses `../kvs_log_configuration`, which would be the SDK-provided default if the working directory was `build`, but not in the case of using the `kvssink` plugin (unless you're also running `gst-launch-1.0` while in `build`). Putting a copy of `kvs_log_configuration` in `..`, relative to whatever your working directory may be, will cause the debug logs to go away.

Also, using the `log-config` property of the `kvssink`, even setting it to the path to the SDK-provided file, would work here.

**Reply by colonelwatch** [reply]
Oh, some further clarity can be added here, if you look at the code linked below, the configuration file is loaded on the `kvssink` element's transition from the NULL state--so before any upload. That overrides the hardcode you observed.

https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/blame/7acf272672ced72bbbad85fe39d366deca57280e/src/gstreamer/gstkvssink.cpp#L1584-L1591
